src/wazuh_db/wdb_agent.c

```diff
@@ -409,178 +359,108 @@ int wdb_remove_agent_db(int id, const char * name) {
-    int result = 0;
     int output = -1;
     char wdbquery[OS_BUFFER_SIZE] = "";
-    char wdboutput[OS_MAXSTR] = ""; 
-    char *json_string = NULL;
+    char wdboutput[OS_BUFFER_SIZE] = "";
     cJSON *root = NULL;
-    cJSON *elem = NULL;
-    cJSON *json_name = NULL;
+    cJSON *json_id = NULL;
 
-    sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_FIND_AGENT], name, ip, ip);
-    result = wdbc_query_ex( &wdb_sock_agent, wdbquery, wdboutput, sizeof(wdboutput));
-
-    switch (result){
-        case OS_SUCCESS:
-            break;
-        case OS_INVALID:
-            mdebug1("GLobal DB Error in the response from socket");
-            mdebug2("Global DB SQL query: %s", wdbquery);
-            return OS_INVALID;
-        default:
-            mdebug1("GLobal DB Cannot execute SQL query; err database %s/%s.db", WDB2_DIR, WDB2_GLOB_NAME);
-            mdebug2("Global DB SQL query: %s", wdbquery);
-            return OS_INVALID;
+    if(!name || !ip){
+        mdebug1("Empty agent name or ip when trying to get agent name. Agent: (%s) IP: (%s)", name, ip);
+        return OS_INVALID;
     }
 
-    if(json_string = wstr_chr(wdboutput, ' '), json_string ){
-        *json_string='\0';
-        json_string++;
-    } else{
-        mdebug1("SQLite result has no space. Query: %s",wdbquery);
+    snprintf(wdbquery, sizeof(wdbquery), global_db_queries[SQL_FIND_AGENT], name, ip, ip);
+    root = wdbc_query_parse_json(&wdb_sock_agent, wdbquery, wdboutput, sizeof(wdboutput));
+
+    if (!root) {
+        merror("Error querying Wazuh DB for agent name.");
         return OS_INVALID;
     }
 
-    if(strcmp(wdboutput,"ok") == 0 && strcmp(json_string,"[]") != 0 && strcmp(json_string,"[{}]") != 0){
-        root = cJSON_Parse(json_string);    
-        elem = cJSON_GetArrayItem(root, 0);
-        json_name = cJSON_GetObjectItem(elem, "id");
-        output = json_name->valueint;
-    } else{
-        mdebug1("SQLite Query failed: %s", wdbquery);
-        return OS_INVALID;
+    json_id = cJSON_GetObjectItem(cJSON_GetArrayItem(root, 0),"id");
+    if(json_id){
+        output = json_id -> valueint;
     }
-        
+
+    cJSON_Delete(root);
     return output;
 }
 
-/* Get the file offset. Returns -1 on error or NULL. */
+/* Get the file offset. Returns -1 on error. */
 long wdb_get_agent_offset(int id_agent, int type) {
     long int output = 0;
-    int result = -1;
-    bool is_FIM = FALSE;
     char wdbquery[OS_BUFFER_SIZE] = "";
-    char wdboutput[OS_MAXSTR] = "";
-    char *json_string = NULL;
+    char wdboutput[OS_BUFFER_SIZE] = "";
     cJSON *root = NULL;
-    cJSON *elem = NULL;
-    cJSON *json_name = NULL;
+    cJSON *json_offset = NULL;
+    char * column = NULL;
 
     switch (type) {
     case WDB_SYSCHECK:
         sqlite3_snprintf(sizeof(wdbquery), wdbquery, global_db_queries[SQL_SELECT_FIM_OFFSET], id_agent);
-        is_FIM = TRUE;
+        os_strdup("fim_offset",column);
```

Comment:
`column` should be freed at the end of the method.

Reply:
Done!